runtime/jcl/common/mgmtinit.c

```diff
@@ -502,21 +503,32 @@ gcStartEvent(J9JavaVM *vm, UDATA heapSize, UDATA heapUsed, UDATA *totals, UDATA
 
 	mgmt->preCollectionHeapSize = heapSize;
 	mgmt->preCollectionHeapUsed = heapUsed;
+	gcInfo = &gcData->lastGcInfo;
 
 	for (idx = 0; idx < mgmt->supportedMemoryPools; ++idx) {
 		J9MemoryPoolData *memoryPool = &memoryPools[idx];
 		total = totals[idx];
 		used = totals[idx] - frees[idx];
 
-		memoryPool->preCollectionSize = total;
-		memoryPool->preCollectionUsed = used;
-		memoryPool->preCollectionMaxSize = memoryPool->postCollectionMaxSize;
+		/* update pre Memory Usage of last GcInfo for the collector */
+		gcInfo->preUsed[idx] = used;
```

Comment:
why doesn't the `memorypool` `pre*` values get updated here anymore?  The peak values till get updated below

Reply:
@LinHu2016 Can you comment on the above question?  The old code was setting `memoryPool->preCollectionSize` here but the new code isn't

Reply:
the pre values has been saved to gcInfo->pre* in stead of updating memoryPool->pre*, because only if current gc manages the memory pool, we can update the memoryPool Collection values (such as the scavenge should not change tenure memorypool collection values, the scavenge does not reclaim memory from tenure pool), we do update collection values from gcInfo in gcEndEvent() (line762-774) for the collector managed pools.